### PR TITLE
Make Maps.toUnmodifiableSortedMap return NavigableMap

### PR DESCRIPTION
**Pull request by arteam**
`Maps.toUnmodifiableSortedMap` was recently introduced  in #77116. We can improve it by returning `NavigableMap` rather than `SortedMap`. `NavigableMap` is a richer interface than `SortedMap` and used more often in the ES codebase. By changing the method to return `NavigableMap`, we can use it in more places without changing a consumer's interface.

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/common/util/Maps.java [hunk 1]
```diff
@@ -199,13 +199,13 @@
      * unmodifiable sorted map. The resulting read-only view through the unmodifiable sorted map is a sorted map.
      *
      * @param <T> the type of the input elements
-     * @return an unmodifiable map where the underlying map is sorted
+     * @return an unmodifiable {@link NavigableMap} where the underlying map is sorted
      */
-    public static <T, K, V> Collector<T, ?, SortedMap<K, V>> toUnmodifiableSortedMap(Function<T, ? extends K> keyMapper,
-                                                                                     Function<T, ? extends V> valueMapper) {
+    public static <T, K, V> Collector<T, ?, NavigableMap<K, V>> toUnmodifiableSortedMap(Function<T, ? extends K> keyMapper,
```
Comment (arteam) [inline server/src/main/java/org/elasticsearch/common/util/Maps.java:204]: Let me know whether it makes sense to rename the method to `toUnmodifiableSortedMap`

**Comment by arteam** [comment]
@elasticmachine update branch

**Comment by arteam** [comment]
@elasticmachine update branch

**Review by original-brownbear — COMMENTED** [review]
Do you have any concrete spots where this change would be helpful in mind? Maybe adjust those while doing this change?

**Comment by arteam** [comment]
@elasticmachine
run elasticsearch-ci/bwc
run elasticsearch-ci/part-2
run elasticsearch-ci/rest-compatibility

**Comment by arteam** [comment]
@elasticmachine update branch

**Comment by arteam** [comment]
@elasticmachine  run elasticsearch-ci/packaging-tests-unix-sample

**Comment by arteam** [comment]
@elasticmachine update branch

**Comment by arteam** [comment]
@elasticmachine update branch

**Comment by arteam** [comment]
@original-brownbear @martijnvg @tlrx Hi guys, could somebody from you guys take a look at this PR? Github suggested you as potential reviewers.

File: server/src/main/java/org/elasticsearch/common/util/Maps.java (hunk 1 above)
Comment (tlrx) [reply server/src/main/java/org/elasticsearch/common/util/Maps.java:204]: Yes, I think it makes sense

**Review by tlrx — APPROVED** [review]
LGTM

**Review by original-brownbear — APPROVED** [review]

**Review by martijnvg — APPROVED** [review]
LGTM

**Comment by arteam** [comment]
Thanks you both!

**Pull request merged**